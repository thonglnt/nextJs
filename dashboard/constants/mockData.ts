import { ROUTE } from './route'

import product1 from '../public/images/products/product_1.png'
import product2 from '../public/images/products/product_2.png'
import product3 from '../public/images/products/product_3.png'
import product4 from '../public/images/products/product_4.png'
import product5 from '../public/images/products/product_5.png'
import product from '../public/images/products/product-1.png'

export const MOCK_USERS = {
  id: '1',
  name: 'John Wick',
  avatar: 'avatar',
  email: 'email123@gmail.com',
  password: '1234Abcd',
}

export const MOCK_LIST_USERS = [
  {
    id: '1',
    name: 'John Wick',
    avatar: 'avatar',
    email: 'email123@gmail.com',
    password: '1234Abcd',
  },
  {
    id: 'o',
    name: 'Johnny',
    avatar: 'avatar',
    email: 'email1234@gmail.com',
    password: '1234Abcd',
  },
]

export const MOCK_SIDEBAR = [
  {
    id: 'GEN',
    title: 'general',
    menu: [
      {
        id: 'GEN_1',
        icon: 'home',
        title: 'Overview',
        slug: ROUTE.HOME_PAGE,
        subMenu: [],
      },
      {
        id: 'GEN_2',
        icon: 'chartBar',
        title: 'Analytics',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'GEN_3',
        icon: 'chartPie',
        title: 'Finance',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'GEN_4',
        icon: 'truck',
        title: 'Logistics',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'GEN_5',
        icon: 'user',
        title: 'Account',
        slug: ROUTE.ACCOUNT,
        subMenu: [],
      },
    ],
  },
  {
    id: 'MAN',
    title: 'management',
    menu: [
      {
        id: 'MAN_1',
        icon: 'users',
        title: 'Customers',
        subMenu: [
          {
            id: 'CUS_LIST',
            title: 'list',
            slug: ROUTE.CUSTOMER,
          },
          {
            id: 'CUS_DETAIL',
            title: 'details',
            slug: `${ROUTE.CUSTOMER}/detail`,
          },
          {
            id: 'CUS_EDIT',
            title: 'edit',
            slug: `${ROUTE.CUSTOMER}/edit`,
          },
        ],
      },
      {
        id: 'MAN_2',
        icon: 'shoppingBag',
        title: 'Products',
        subMenu: [
          {
            id: 'PROD_LIST',
            title: 'list',
            slug: ROUTE.PRODUCTS,
          },
          {
            id: 'PROD_CREATE',
            title: 'create',
            slug: `${ROUTE.PRODUCTS}/create`,
          },
        ],
      },
      {
        id: 'MAN_3',
        icon: 'shoppingCart',
        title: 'Orders',
        subMenu: [
          {
            id: 'ORD_LIST',
            title: 'list',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'ORD_DETAIL',
            title: 'details',
            slug: ROUTE.DEFAULT,
          },
        ],
      },
      {
        id: 'MAN_4',
        icon: 'receipt',
        title: 'Invoices',
        subMenu: [
          {
            id: 'INV_LIST',
            title: 'list',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'INV_DETAIL',
            title: 'details',
            slug: ROUTE.DEFAULT,
          },
        ],
      },
    ],
  },
  {
    id: 'PLAT',
    title: 'platforms',
    menu: [
      {
        id: 'PLAT_1',
        icon: 'officeBuilding',
        title: 'Job Listings',
        subMenu: [
          {
            id: 'JOB_1',
            title: 'browse',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'JOB_2',
            title: 'details',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'JOB_3',
            title: 'create',
            slug: ROUTE.DEFAULT,
          },
        ],
      },
      {
        id: 'PLAT_2',
        icon: 'share',
        title: 'Social Media',
        subMenu: [
          {
            id: 'SOCIAL_1',
            title: 'profile',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'SOCIAL_2',
            title: 'feed',
            slug: ROUTE.DEFAULT,
          },
        ],
      },
      {
        id: 'PLAT_3',
        icon: 'blog',
        title: 'Blog',
        subMenu: [
          {
            id: 'BLOG_1',
            title: 'post list',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'BLOG_2',
            title: 'post details',
            slug: ROUTE.DEFAULT,
          },
          {
            id: 'BLOG_3',
            title: 'post create',
            slug: ROUTE.DEFAULT,
          },
        ],
      },
    ],
  },
  {
    id: 'APP',
    title: 'apps',
    menu: [
      {
        id: 'KANBAN',
        icon: 'clipboard',
        title: 'kanban',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'MAIL',
        icon: 'mail',
        title: 'mail',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'CHAT',
        icon: 'chat',
        title: 'Chat',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'CALENDAR',
        icon: 'calendar',
        title: 'Calendar',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
    ],
  },
  {
    id: 'PAGES',
    title: 'pages',
    menu: [
      {
        id: 'AUTH',
        icon: 'lockClose',
        title: 'Auth',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'PRICING',
        icon: 'creditCard',
        title: 'Pricing',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'CHECKOUT',
        icon: 'cash',
        title: 'Checkout',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'CONTACT',
        icon: 'mail',
        title: 'Contact',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
      {
        id: 'ERR',
        icon: 'xCircle',
        title: 'Error',
        slug: ROUTE.DEFAULT,
        subMenu: [],
      },
    ],
  },
]

export const MOCK_PRODUCTS = [
  {
    id: 1,
    name: 'Dropbox',
    imageUrl: product1,
    updatedAt: 2,
  },
  {
    id: 2,
    name: 'Medium Corporation',
    imageUrl: product2,
    updatedAt: 2,
  },
  {
    id: 3,
    name: 'Slack',
    imageUrl: product3,
    updatedAt: 3,
  },
  {
    id: 4,
    name: 'Lyft',
    imageUrl: product4,
    updatedAt: 5,
  },
  {
    id: 5,
    name: 'GitHub',
    imageUrl: product5,
    updatedAt: 9,
  },
]

export const MOCK_ORDERS = [
  {
    id: 1,
    ref: 'CDD1049',
    amount: 30.5,
    customer: {
      name: 'Ekaterina Tankova',
    },
    createdAt: '12/04/2019',
    status: 'pending',
  },
  {
    id: 2,
    ref: 'CDD1048',
    amount: 25.1,
    customer: {
      name: 'Cao Yu',
    },
    createdAt: '12/04/2019',
    status: 'delivered',
  },
  {
    id: 3,
    ref: 'CDD1047',
    amount: 10.99,
    customer: {
      name: 'Alexa Richardson',
    },
    createdAt: '12/04/2019',
    status: 'refunded',
  },
  {
    id: 4,
    ref: 'CDD1046',
    amount: 96.43,
    customer: {
      name: 'Anje Keizer',
    },
    createdAt: '12/04/2019',
    status: 'pending',
  },
  {
    id: 5,
    ref: 'CDD1045',
    amount: 32.54,
    customer: {
      name: 'Clarke Gillebert',
    },
    createdAt: '12/04/2019',
    status: 'delivered',
  },
  {
    id: 6,
    ref: 'CDD1044',
    amount: 16.76,
    customer: {
      name: 'Adam Denisov',
    },
    createdAt: '12/04/2019',
    status: 'delivered',
  },
]

export const MOCK_INVOICE = [
  {
    id: 1,
    date: '2 Jun 2021',
    total: 4.99,
  },
  {
    id: 2,
    date: '2 Jun 2021',
    total: 4.99,
  },
  {
    id: 3,
    date: '2 Jun 2021',
    total: 4.99,
  },
]

export const MOCK_LOGIN_HISTORY = [
  {
    id: 1,
    type: 'on 10:40 AM 2021/09/01',
    ip: '95.130.17.84',
    client: 'Chrome, Mac OS 10.15.7',
  },
  {
    id: 2,
    type: 'on 10:40 AM 2021/09/01',
    ip: '95.130.17.84',
    client: 'Chrome, Mac OS 10.15.7',
  },
]

export const MOCK_DATA_CUSTOMERS = [
  {
    name: 'Ekaterina Tankova',
    location: 'Parkersburg, West Virginia, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Cao Yu',
    location: 'Iowa, Bristow, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Alexa Richardson',
    location: 'Atlanta, Georgia, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Anje Keizer',
    location: 'Dover, Ohio, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Clarke Gillebert',
    location: 'Dallas, Texas, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Adam Denisov',
    location: 'Bakerfield, California, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Ava Gregoraci',
    location: 'Redondo Beach, California, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Emilee Simchenko',
    location: 'Las Vegas, Nevada, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Kwak Seong-Min',
    location: 'Detroit, Michigan, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
  {
    name: 'Merrile Burgett',
    location: 'Salt Lake City, Utah, USA',
    orders: 3,
    spends: 300,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    email: 'carson.darrin@devias.io',
  },
]

export const MOCK_DATA_PRODUCTS = [
  {
    name: 'Healthcare Erbology',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 1',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 2',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 3',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 4',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 5',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 6',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 7',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 8',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
  {
    name: 'Healthcare Erbology 9',
    image: product,
    stock: 85,
    price: 23.99,
    sku: '401_1BBXBK',
    status: 'published',
    category: 'healthcare',
    oldPrice: 23.99,
    barcode: '5ece2c077e39da27658aa8a9',
    isActive: false,
  },
]

export const MOCK_LIST_CATEGORY = [
  {
    value: 1,
    title: 'Healthcare',
  },
  {
    value: 2,
    title: 'Makeup',
  },
  {
    value: 3,
    title: 'Dress',
  },
  {
    value: 4,
    title: 'Skincare',
  },
  {
    value: 5,
    title: 'Jewelry',
  },
  {
    value: 6,
    title: 'Blouse',
  },
]

export const MOCK_NOTIFICATIONS = [
  {
    id: 1,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    content: 'Jie Yang Song added a new job',
    link: 'Remote React / React Native Developer',
    time: 'Feb 09, 6:52 AM',
  },
  {
    id: 2,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    content: 'Jie Yang Song added a new job',
    link: 'Senior Golang Backend Engineer',
    time: 'Feb 09, 6:52 AM',
  },
  {
    id: 3,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    content: 'New feature! Logistics management is now available',
    link: '',
    time: 'Feb 09, 6:52 AM',
  },
  {
    id: 4,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    content: 'created',
    link: 'Augmastic Inc',
    time: 'Feb 09, 6:52 AM',
  },
]

export const MOCK_CONTACTS = [
  {
    id: 1,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    name: 'Alcides Antonio',
    active: '',
    online: true,
  },
  {
    id: 2,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    name: 'Marcus Finn',
    active: '3h',
    online: false,
  },
  {
    id: 3,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    name: 'Carson Darrin',
    active: '1h',
    online: false,
  },
  {
    id: 4,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    name: 'Fran Perez',
    active: '',
    online: true,
  },
  {
    id: 5,
    img: 'https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png',
    alt: 'avatar',
    name: 'Jie Yan Song',
    active: '',
    online: true,
  },
]
