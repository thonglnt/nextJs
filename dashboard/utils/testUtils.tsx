import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
