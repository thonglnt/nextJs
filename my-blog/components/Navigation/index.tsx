'use client'

import {
  Box,
  Container,
  IconButton,
  List,
  ListItem,
  SvgIcon,
  useTheme,
} from '@mui/material'
import Link from 'next/link'
import { lazy, Suspense, useState } from 'react'
import SearchIcon from '@mui/icons-material/Search'
import MenuIcon from '@mui/icons-material/Menu'

// Components
import Button from '@components/Button'

const Search = lazy(() => import('@components/Search'))

type TNavItem = {
  id: number
  name: string
  path: string
}

export interface INavigation {
  navItems: TNavItem[]
}

const Navigation = ({ navItems }: INavigation) => {
  const theme = useTheme()
  const [isOpen, setIsOpen] = useState(false)
  const [isOpenNavigate, setIsOpenNavigate] = useState(false)

  const handleOpen = () => {
    setIsOpen(true)
  }

  const handleClose = () => {
    setIsOpen(false)
  }

  const handleGetValue = () => {
    // console.log(123)
  }

  const handleKeyDown = () => {
    // console.log(123)
  }

  const handleOpenNavigate = () => {
    if (isOpenNavigate) {
      setIsOpenNavigate(false)
    } else {
      setIsOpenNavigate(true)
    }
  }

  return (
    <>
      <Box
        display="flex"
        justifyContent="space-between"
        padding="12px 12px 12px 25px"
        width="100%"
        alignItems="center"
        boxShadow={`${theme.palette.grey[400]} 0px 3px 8px`}
        bgcolor={theme.palette.common.white}
      >
        {/* List Desktop */}
        <List
          sx={{
            display: {
              md: 'flex',
              xs: 'none',
            },
            gap: '30px',
          }}
        >
          {navItems.map((item) => (
            <ListItem
              key={item.id}
              sx={{
                padding: 0,
                fontSize: '18px',
                textTransform: 'uppercase',
                fontWeight: 600,
                letterSpacing: '2px',
                width: 'auto',
                ':hover': {
                  color: theme.palette.grey[500],
                },
              }}
            >
              <Link href={item.path} prefetch={false}>
                {item.name}
              </Link>
            </ListItem>
          ))}
        </List>

        {/* List mobile */}
        <IconButton
          edge="start"
          color="inherit"
          aria-label="menu"
          sx={{
            mr: 2,
            display: {
              md: 'none',
              xs: 'block',
            },
          }}
          onClick={handleOpenNavigate}
        >
          <MenuIcon />
        </IconButton>

        {/* Search Button */}
        <Button
          onClick={handleOpen}
          variant="text"
          aria-label="search"
          sx={{
            backgroundColor: 'transparent',
            ':hover': {
              backgroundColor: 'transparent',
            },
          }}
        >
          <SvgIcon
            sx={{
              color: theme.palette.common.black,
              width: 24,
              height: 24,
              cursor: 'pointer',
              ':hover': {
                color: theme.palette.grey[500],
              },
            }}
          >
            <SearchIcon />
          </SvgIcon>
        </Button>
      </Box>

      {/* Modal Search */}
      <Suspense fallback={<div>loading.....</div>}>
        <Search
          isOpen={isOpen}
          onClose={handleClose}
          onChange={handleGetValue}
          onKeyDown={handleKeyDown}
        />
      </Suspense>

      {/* Modal Navigate */}
      {isOpenNavigate && (
        <Container
          sx={{
            position: 'relative',
          }}
        >
          <Box
            width="100%"
            sx={{
              display: {
                md: 'none',
                xs: 'block',
              },
              zIndex: 2,
            }}
            bgcolor={theme.palette.grey[800]}
            position="absolute"
            right="0"
          >
            <List
              sx={{
                gap: '12px',
                display: 'flex',
                flexDirection: 'column',
              }}
            >
              {navItems.map((item) => (
                <ListItem
                  key={item.id}
                  sx={{
                    padding: 0,
                    fontSize: '16px',
                    textTransform: 'uppercase',
                    fontWeight: 600,
                    letterSpacing: '2px',
                    width: 'auto',
                    ':hover': {
                      bgcolor: theme.palette.grey[500],
                    },
                    color: theme.palette.common.white,
                  }}
                >
                  <Link
                    href={item.path}
                    prefetch={false}
                    style={{
                      width: '100%',
                      padding: '15px 25px',
                    }}
                  >
                    {item.name}
                  </Link>
                </ListItem>
              ))}
            </List>
          </Box>
        </Container>
      )}
    </>
  )
}

export default Navigation
