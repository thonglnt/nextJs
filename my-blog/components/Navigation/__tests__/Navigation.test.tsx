// Utils
import { fireEvent, render, screen } from '@utils/testUtils'

// Mocks data
import { MOCK_NAVIGATE } from '@constants/__mock__/mockData'

// Components
import Navigation, { INavigation } from '..'

describe('Navigation Component', () => {
  const mockProps: INavigation = MOCK_NAVIGATE

  jest.mock('next/router', () => ({
    useRouter: () => ({
      push: jest.fn(),
      back: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn(),
      },
    }),
  }))

  it('Should show match Navigation DOM Snapshot ', () => {
    const { container } = render(<Navigation {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render math content Navigation', () => {
    render(<Navigation {...mockProps} />)

    const home = screen.getByRole('link', {
      name: /home/i,
    })
    const mostView = screen.getByRole('link', {
      name: /Most View/i,
    })
    const blog = screen.getByRole('link', {
      name: /Blog/i,
    })
    const about = screen.getByRole('link', {
      name: /About/i,
    })

    expect(home).toHaveTextContent('Home')
    expect(mostView).toHaveTextContent('Most View')
    expect(blog).toHaveTextContent('Blog')
    expect(about).toHaveTextContent('About')

    expect(home).toHaveAttribute('href', '/')
    expect(mostView).toHaveAttribute('href', '/most-view/1')
    expect(blog).toHaveAttribute('href', '/blog/1')
    expect(about).toHaveAttribute('href', '/about')
  })

  it('handleOpen should call and open modal search', () => {
    render(<Navigation {...mockProps} />)

    const buttonSearch = screen.getByRole('button', {
      name: /search/i,
    })

    fireEvent.click(buttonSearch)

    expect(screen.getByTestId('backdrop')).toBeInTheDocument()
  })

  it('handleClose should call and close modal search', () => {
    const { container } = render(<Navigation {...mockProps} />)

    const buttonSearch = screen.getByRole('button', {
      name: /search/i,
    })

    fireEvent.click(buttonSearch)

    const buttonClose = screen.getByTestId('close-icon')

    fireEvent.click(buttonClose)

    expect(container).not.toBe('Type to search')
  })

  it('handleGetValue should call and get correct value', () => {
    render(<Navigation {...mockProps} />)

    const buttonSearch = screen.getByRole('button', {
      name: /search/i,
    })

    fireEvent.click(buttonSearch)

    const search = screen.getByRole('textbox') as HTMLInputElement

    fireEvent.change(search, {
      target: { value: 'Search Value' },
    })

    expect(search.value).toBe('Search Value')
  })

  it('handleKeyDown should call and get correct value', () => {
    render(<Navigation {...mockProps} />)

    const buttonSearch = screen.getByRole('button', {
      name: /search/i,
    })

    fireEvent.click(buttonSearch)

    const search = screen.getByRole('textbox')
    fireEvent.keyDown(search)

    expect(search).toBeTruthy()
  })

  it('handleOpenNavigate should call and open menu navigation on mobile', () => {
    render(<Navigation {...mockProps} />)

    const buttonMenu = screen.getByRole('button', {
      name: /menu/i,
    })

    fireEvent.click(buttonMenu)

    const homeLink = screen.getAllByRole('link', {
      name: /home/i,
    })

    const mostViewLink = screen.getAllByRole('link', {
      name: /most view/i,
    })

    const blogLink = screen.getAllByRole('link', {
      name: /blog/i,
    })

    const aboutLink = screen.getAllByRole('link', {
      name: /about/i,
    })

    expect(homeLink[1]).toBeTruthy()
    expect(mostViewLink[1]).toBeTruthy()
    expect(blogLink[1]).toBeTruthy()
    expect(aboutLink[1]).toBeTruthy()
  })

  it('handleOpenNavigate should call and open menu navigation on mobile', () => {
    render(<Navigation {...mockProps} />)

    const buttonMenu = screen.getByRole('button', {
      name: /menu/i,
    })

    fireEvent.click(buttonMenu)
    fireEvent.click(buttonMenu)

    const homeLink = screen.getAllByRole('link', {
      name: /home/i,
    })

    expect(homeLink[1]).not.toBeTruthy()
  })
})
