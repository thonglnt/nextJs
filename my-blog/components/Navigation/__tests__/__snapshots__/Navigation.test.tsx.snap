// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component Should show match Navigation DOM Snapshot  1`] = `
<div>
  <div
    class="MuiBox-root css-7w1jqr"
  >
    <ul
      class="MuiList-root MuiList-padding css-127t676-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
      >
        <a
          href="/most-view/1"
        >
          Most View
        </a>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
      >
        <a
          href="/blog/1"
        >
          Blog
        </a>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
      >
        <a
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
    <button
      aria-label="menu"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-bxr13g-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="search"
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-si6aep-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1unfims-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    loading.....
  </div>
</div>
`;
